joshprzybyszewski/cribbage
# Bake the React client into the golang gin server

## What broke / What you're adding
The React client exists (even if it doesn't really work:#) and it should be served by the gin server. 

## How you did it
This will build the React app into its binary and then serve that static file from the gin server. I followed the examples in [this blog](https://medium.com/@synapticsynergy/serving-a-react-app-with-golang-using-gin-c6402ee64a4b). And I referenced [this blog](https://mherman.org/blog/dockerizing-a-react-app/), this ["getting started" guide](https://create-react-app.dev/docs/getting-started/), grabbed the node docker image [here](https://hub.docker.com/_/node?ref=login&tab=description). And I opened a docker hub [repo for cribbage](https://hub.docker.com/r/joshprzybyszewski/cribbage).

## How to test it and how to try to break it
run `make dockerbuild dockerrunlocal` and see the react app when you go to [http://localhost:8081](http://localhost:8081).
